lib/linter/code-path-analysis/code-path-analyzer.js

```diff
@@ -244,6 +244,17 @@ function preprocess(analyzer, node) {
     const parent = node.parent;
 
     switch (parent.type) {
+        case "CallExpression":
+            if (parent.optional === true && parent.arguments.length >= 1 && parent.arguments[0] === node) {
+                state.makeOptionalRight();
+            }
+            break;
+        case "MemberExpression":
+            if (parent.optional === true && parent.property === node) {
+                state.makeOptionalRight();
+            }
+            break;
```

Comment:
These might be missing cases where there's an optional call without arguments in the chain?

`a?.().b;`

![image](https://user-images.githubusercontent.com/44349756/85951897-5f022400-b966-11ea-8652-5d30a6ccda80.png)

there should be fork from `Identifier(a)`?

Similar: `a?.()(b);`

Reply:
Good catch!